modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/webapp/mgt/config/WebAppDescriptorTestBase.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.appserver.integration.tests.webapp.mgt.config;
+
+import org.testng.Assert;
+import org.wso2.appserver.integration.common.clients.WebAppAdminClient;
+import org.wso2.appserver.integration.common.utils.ASIntegrationTest;
+import org.wso2.appserver.integration.common.utils.WebAppDeploymentUtil;
+import org.wso2.appserver.integration.common.utils.WebAppMode;
+import org.wso2.carbon.automation.engine.exceptions.AutomationFrameworkException;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpClientUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import java.nio.file.Paths;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class WebAppDescriptorTestBase extends ASIntegrationTest {
+    public static final String PASS = "Pass";
+    public static final String FAIL = "Fail";
+    public static final String SAMPLE_APP_LOCATION = Paths
+            .get(FrameworkPathUtil.getSystemResourceLocation() + "artifacts", "AS", "war", "webAppDescriptorTesting")
+            .toString();
+    protected final String webAppFileName;
+    protected final String webAppName;
+    protected final String webAppLocalURL;
+    protected final String hostName = "localhost";
+    protected String sampleAppDirectory;
+    protected WebAppAdminClient webAppAdminClient;
+
+    public WebAppDescriptorTestBase(WebAppMode webAppMode) {
+        webAppName = webAppMode.getWebAppName().split(".war")[0];
+        webAppFileName = webAppMode.getWebAppName();
+        webAppLocalURL = "/" + webAppName;
+
+    }
+
+    protected void init() throws Exception {
+        super.init();
+        webAppAdminClient = new WebAppAdminClient(backendURL, sessionCookie);
+    }
+
+    protected void webApplicationDeployment() throws Exception {
```

Comment:
That is because this is a base class, the methods of which are being used by the test classes extending this.